src/lavinmq/exchange/exchange.cr

```diff
@@ -105,22 +108,12 @@ module LavinMQ
     end
 
     def in_use?
-      return true if @queue_bindings.size > 0
-      return true if @exchange_bindings.size > 0
-      return true if @vhost.exchanges.any? { |_, x| x.exchange_bindings.any? { |_, exs| exs.includes? self } }
-      false
-    end
-
-    def bindings_details
-      Iterator.chain({@queue_bindings.each, @exchange_bindings.each}).flat_map do |(key, destinations)|
-        destinations.map { |destination| binding_details(key, destination) }
+      return true unless bindings_details.empty?
+      @vhost.exchanges.any? do |_, x|
+        x.bindings_details.each.any? { |bd| bd.destination == self }
```

Comment:
```suggestion
        x.bindings_details.any? { |bd| bd.destination == self }
```

Reply:
`.any?` does only iterate over the hash, doesn't generate a new array or anything, so the extra `.each` isn't necessary.